requests-for-comments/2022-11-08-supported-languages.md

```diff
@@ -0,0 +1,169 @@
+# Request for comments: Supported languages at PostHog
+
+Since day 1 here at Posthog we've supported two and half core languages: Python and Javascript/Typescript.
+
+This set of tools has carried us a long way and we should both the cpython runtime and node a huge hats off for bringing us this far.
+
+## Problem statement
+
+We have the opportunity coming up to rebuild or greenfield build out services that are critical parts of our data pipeline. It will be important for these services to be correct, fast, and efficient. Considering this now is a good time to ask: Are we using the correct tools for the job?
+
+So why even bring this up?
+
+Frankly:
+
+- Python is slow.
+- Node is a memory hog.
+- Node tooling is slow.
+- Dependencies can be huge.
+- No guarantees that code is correct.
+
+Can we be successful continuing to use these languages? Of course.
+
+- They are well understood.
+- Our tooling is setup to support these languages
+- We have internal expertise.
+- Generally they are _good enough_.
+
+Why should we be open to other languages?
+
+- We have different people with experience working with different languages.
+- 100% guaranteed static typing is quite beneficial
+- Compiled languages provide more confidence shipped code is correct
+- There are compiled languages that provide significant efficiency wins on CPU, Memory, and performance
+- Using the right tool for the job is typically the right thing to do (if you can ship it)
+
+Candidate Languages:
+
+- [Golang](https://go.dev/) (big surprise)
+- [Rust](https://www.rust-lang.org/) 🦀
+- [OCaml](https://ocaml.org/)
+- [Elixir](https://elixir-lang.org/) (dynamically typed)
+- [Scala](https://www.scala-lang.org/)
+- [Java](<https://en.wikipedia.org/wiki/Java_(programming_language)>)
+
+## Meet the eligible languages
+
+### Golang
```

Comment:
Tbh I would not be against us introducing Go for a new service, so long as we're sure that Python does not and cannot suit our use case